apps/studio/components/interfaces/Integrations/Wrappers/CreateDuckDbWrapperSheet.tsx

```diff
@@ -0,0 +1,499 @@
+
+  const [formErrors, setFormErrors] = useState<{ [k: string]: string }>({})
+
+  const { mutateAsync: createFDW, isLoading: isCreatingWrapper } = useFDWCreateMutation({
+    onSuccess: () => {
+      toast.success(`Successfully created ${wrapperMeta?.label} foreign data wrapper`)
+      onClose()
+    },
+  })
+
+  const wrapperMeta = useMemo(() => {
+    const fields = requiredFields[selectedTarget]
+
+    return {
+      ...wrapperMetaOriginal,
+      server: {
+        options: wrapperMetaOriginal.server.options
+          // when the target changes, change the wrapperMeta field
+          .filter((option) => fields.find((field) => field.name === option.name))
+          .map((option) => {
+            const field = fields.find((field) => field.name === option.name)
+            return {
+              ...option,
+              required: field?.required ?? false,
+            }
+          }),
+      },
+    }
+  }, [wrapperMetaOriginal, selectedTarget])
+
+  const { data: schemas } = useSchemasQuery({
+    projectRef: project?.ref!,
+    connectionString: project?.connectionString,
+  })
+
+  const initialValues = {
+    wrapper_name: '',
+    server_name: '',
+    source_schema: wrapperMeta.sourceSchemaOption?.defaultValue ?? '',
+    target_schema: '',
+    ...Object.fromEntries(
+      wrapperMeta.server.options.map((option) => [option.name, option.defaultValue ?? ''])
+    ),
+  }
+
+  const { mutateAsync: createSchema, isLoading: isCreatingSchema } = useSchemaCreateMutation()
+
+  const onSubmit = async (values: any) => {
+    const validate = makeValidateRequired(wrapperMeta.server.options)
+    const errors: any = validate(values)
+
+    if (selectedTarget === 's3') {
+      if (
+        values.url_compatibility_mode &&
+        values.url_compatibility_mode !== 'true' &&
+        values.url_compatibility_mode !== 'false'
+      ) {
+        errors.url_compatibility_mode = 'Please set it true or false'
+      }
+
+      if (values.url_style && values.url_style !== 'vhost' && values.url_style !== 'path') {
+        errors.url_style = 'Please set it vhost or path'
+      }
+
+      if (values.use_ssl && values.use_ssl !== 'true' && values.use_ssl !== 'false') {
+        errors.use_ssl = 'Please set it true or false'
+      }
+    }
+
+    if (values.source_schema.length === 0) {
+      errors.source_schema = 'Please provide a namespace name'
+    }
+    if (values.wrapper_name.length === 0) {
+      errors.wrapper_name = 'Please provide a name for your wrapper'
+    }
+
+    if (values.target_schema.length === 0) {
+      errors.target_schema = 'Please provide an unique target schema'
+    }
+    const foundSchema = schemas?.find((s) => s.name === values.target_schema)
+    if (foundSchema) {
+      errors.target_schema = 'This schema already exists. Please specify a unique schema name.'
+    }
+
+    setFormErrors(errors)
+    if (!isEmpty(errors)) {
+      return
+    }
+
+    try {
+      await createSchema({
```

Comment:
If the FDW creation failed, the schema creation should ideally be rolled back. Otherwise, the user can't try again with the same schema name without deleting their schema